app/models/agents/rss_agent.rb

```diff
@@ -160,7 +165,9 @@ def check_and_track(entry_id)
         false
       else
         memory['seen_ids'].unshift entry_id
-        memory['seen_ids'].pop if memory['seen_ids'].length > 500
+        while memory['seen_ids'].length > ((interpolated['max_ids'].presence || 500).to_i) do
```

Comment:
You can remove the loop by passing an amount to [`pop`](https://ruby-doc.org/core-2.2.0/Array.html#method-i-pop).